# Capstone Project - The Battle of Neighborhoods (First Week / Course Week 4)

## Data Section <a name="data"></a>

#### Import necessary Libraries

In [1]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.0

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.

Folium installed
Libraries imported.


#### Define Foursquare Credentials and Version

In [2]:
CLIENT_ID = '2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH' # your Foursquare ID
CLIENT_SECRET = 'FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 10
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH
CLIENT_SECRET:FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS


In [3]:
# Use geopy library to get the latitude and longitude values of Downtown Toronto

address = 'Downtown Toronto, ON'
geolocator = Nominatim(user_agent="Downtown_Toronto_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Downtown Toronto are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Downtown Toronto are 43.6563221, -79.3809161.


<br>

### The Search for a specific venue

Imagine our targeted audience (Cuisine Connoisseurs, Food Enthusiast and Bloggers) is in a downtown area using the Restaurant Guide App and wanted to find great budget-friendly restaurants. It is a busy town, a quick comparison of several venues in close proximity would be ideal for they're always on the go. 
<br>

In [4]:
# define a query to search for Restaurant within 500 metres from Downtown Toronto

search_query = 'Restaurant'
radius = 500
print(search_query + ' .... OK!')

Restaurant .... OK!


#### Define the corresponding URL and limit = 10

In [5]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH&client_secret=FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS&ll=43.6563221,-79.3809161&v=20180604&query=Restaurant&radius=500&limit=10'

#### Make the call to foursquare api; send GET Request and examine results

In [6]:
results = requests.get(url).json()
results

# To simplify, output was cleared

{'meta': {'code': 200, 'requestId': '5df08de8ad1ab4001ba78a45'},
 'response': {'venues': [{'id': '4ad4c05ff964a52048f720e3',
    'name': 'Hemispheres Restaurant & Bistro',
    'location': {'address': '110 Chestnut Street',
     'lat': 43.65488413420439,
     'lng': -79.38593077371578,
     'labeledLatLngs': [{'label': 'display',
       'lat': 43.65488413420439,
       'lng': -79.38593077371578}],
     'distance': 434,
     'postalCode': 'M5G 1R3',
     'cc': 'CA',
     'city': 'Toronto',
     'state': 'ON',
     'country': 'Canada',
     'formattedAddress': ['110 Chestnut Street',
      'Toronto ON M5G 1R3',
      'Canada']},
    'categories': [{'id': '4bf58dd8d48988d14e941735',
      'name': 'American Restaurant',
      'pluralName': 'American Restaurants',
      'shortName': 'American',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1576046072',
    'hasPerk': False},
   {'id': '

#### Get relevant part of JSON and transform it into a *pandas* dataframe

In [7]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

id                                name  \
0  4ad4c05ff964a52048f720e3     Hemispheres Restaurant & Bistro   
1  4edbccaaf9abeea3db5268df       Terrace Restaurant at Elmwood   
2  4c6abb8c2c29d13ad2720e41                              Subway   
3  4ad7929cf964a520500c21e3              The Senator Restaurant   
4  4fbbb3ace4b01e39077b4d55  Akashiro Japanese Restaurant & Bar   

                                          categories    referralId  hasPerk  \
0  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...  v-1576046072    False   
1  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...  v-1576046072    False   
2  [{'id': '4bf58dd8d48988d1c5941735', 'name': 'S...  v-1576046072    False   
3  [{'id': '4bf58dd8d48988d147941735', 'name': 'D...  v-1576046072    False   
4  [{'id': '4bf58dd8d48988d1d2941735', 'name': 'S...  v-1576046072    False   

      location.address  location.lat  location.lng  \
0  110 Chestnut Street     43.654884    -79.385931   
1                  NaN     43.656101    -79.381480   
2      10 Dundas St. E     43.656260    -79.380920   
3  249 Victoria Street     43.655641    -79.379199   
4        220 Yonge St.     43.655965    -79.380541   

                             location.labeledLatLngs  location.distance  \
0  [{'label': 'display', 'lat': 43.65488413420439...                434   
1  [{'label': 'display', 'lat': 43.65610066909799...                 51   
2  [{'label': 'display', 'lat': 43.65625998968890...                  6   
3  [{'label': 'display', 'lat': 43.65564091455335...                157   
4  [{'label': 'display', 'lat': 43.65596504293232...                 49   

  location.postalCode location.cc location.city location.state  \
0             M5G 1R3          CA       Toronto             ON   
1                 NaN          CA           NaN            NaN   
2             M5B 2G9          CA       Toronto             ON   
3             M5B 1T8          CA       Toronto             ON   
4                 NaN          CA       Toronto             ON   

  location.country                          location.formattedAddress  \
0           Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
1           Canada                                           [Canada]   
2           Canada  [10 Dundas St. E (at Yonge St.), Toronto ON M5...   
3           Canada  [249 Victoria Street (btwn Dundas St E and Shu...   
4           Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   

                   location.crossStreet venuePage.id  
0                                   NaN          NaN  
1                                   NaN          NaN  
2                          at Yonge St.          NaN  
3        btwn Dundas St E and Shuter St     55585058  
4  in Toronto Eaton Centre Urban Eatery          NaN

#### Define information of interest and filter dataframe

In [8]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name           categories  \
0      Hemispheres Restaurant & Bistro  American Restaurant   
1        Terrace Restaurant at Elmwood  American Restaurant   
2                               Subway       Sandwich Place   
3               The Senator Restaurant                Diner   
4   Akashiro Japanese Restaurant & Bar     Sushi Restaurant   
5  Richtree Natural Market Restaurants           Restaurant   
6        Hong Shing Chinese Restaurant   Chinese Restaurant   
7                     Adega Restaurant           Restaurant   
8                                 JOEY  American Restaurant   
9      Kyoto House Japanese Restaurant     Sushi Restaurant   

               address        lat        lng  \
0  110 Chestnut Street  43.654884 -79.385931   
1                  NaN  43.656101 -79.381480   
2      10 Dundas St. E  43.656260 -79.380920   
3  249 Victoria Street  43.655641 -79.379199   
4        220 Yonge St.  43.655965 -79.380541   
5        14 Queen St W  43.652614 -79.380231   
6      195 Dundas St W  43.654925 -79.387089   
7            33 Elm St  43.657519 -79.383462   
8     1 Dundas St West  43.656094 -79.381878   
9  143 Dundas St. West  43.655381 -79.385270   

                                      labeledLatLngs  distance postalCode  cc  \
0  [{'label': 'display', 'lat': 43.65488413420439...       434    M5G 1R3  CA   
1  [{'label': 'display', 'lat': 43.65610066909799...        51        NaN  CA   
2  [{'label': 'display', 'lat': 43.65625998968890...         6    M5B 2G9  CA   
3  [{'label': 'display', 'lat': 43.65564091455335...       157    M5B 1T8  CA   
4  [{'label': 'display', 'lat': 43.65596504293232...        49        NaN  CA   
5  [{'label': 'display', 'lat': 43.65261436174172...       416    M5H 3X4  CA   
6  [{'label': 'display', 'lat': 43.65492521335936...       520    M5G 1C7  CA   
7  [{'label': 'display', 'lat': 43.65751886446546...       244    M5G 1H1  CA   
8  [{'label': 'display', 'lat': 43.6560936540828,...        81    M5G 1Z3  CA   
9  [{'label': 'display', 'lat': 43.65538110598594...       365        NaN  CA   

      city state country                                   formattedAddress  \
0  Toronto    ON  Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
1      NaN   NaN  Canada                                           [Canada]   
2  Toronto    ON  Canada  [10 Dundas St. E (at Yonge St.), Toronto ON M5...   
3  Toronto    ON  Canada  [249 Victoria Street (btwn Dundas St E and Shu...   
4  Toronto    ON  Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   
5  Toronto    ON  Canada        [14 Queen St W, Toronto ON M5H 3X4, Canada]   
6  Toronto    ON  Canada  [195 Dundas St W (at University Ave), Toronto ...   
7  Toronto    ON  Canada  [33 Elm St (at Bay St), Toronto ON M5G 1H1, Ca...   
8  Toronto    ON  Canada     [1 Dundas St West, Toronto ON M5G 1Z3, Canada]   
9  Toronto    ON  Canada          [143 Dundas St. West, Toronto ON, Canada]   

                            crossStreet                        id  
0                                   NaN  4ad4c05ff964a52048f720e3  
1                                   NaN  4edbccaaf9abeea3db5268df  
2                          at Yonge St.  4c6abb8c2c29d13ad2720e41  
3        btwn Dundas St E and Shuter St  4ad7929cf964a520500c21e3  
4  in Toronto Eaton Centre Urban Eatery  4fbbb3ace4b01e39077b4d55  
5                                   NaN  4b295e10f964a520ba9d24e3  
6                     at University Ave  4b2027b5f964a520f82d24e3  
7                             at Bay St  4ad4c060f964a52070f720e3  
8                                   NaN  4df909dfe4cd2129701c0690  
9                                   NaN  4ae4b055f964a520229d21e3

In [9]:
dataframe_filtered.name

0        Hemispheres Restaurant & Bistro
1          Terrace Restaurant at Elmwood
2                                 Subway
3                 The Senator Restaurant
4     Akashiro Japanese Restaurant & Bar
5    Richtree Natural Market Restaurants
6          Hong Shing Chinese Restaurant
7                       Adega Restaurant
8                                   JOEY
9        Kyoto House Japanese Restaurant
Name: name, dtype: object

<br>

### Map of The City’s Downtown Area 

Would it be convenient if we could show our targeted audience the map of their search. Also this will show them how far they are from a prospective venue.

In [10]:
# Let’s locate the approximate center and explore neighboring restaurants
# Zoom in to visualize
# Note: Github may not display map

venues_map = folium.Map(location=[latitude, longitude], zoom_start=10) # generate map centred around 

# add a red circle marker to represent 
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Downtown Toronto',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

##### Approximate Defined Center: Google Maps

We searched cordinate 43.6563221, -79.3809161 for Downtown Toronto in google maps. Result will shows "418 The PATH - Yonge-Dundas Square, Toronto, ON M5B 1R4, Canada". This is our approximate defined center and represented by a red circle marker from the generated map above. Using the result, we can test venue location accuracy and other information.

#### Dataframe Clean Up

In [11]:
# Let's sort by distance
sort_by_distance = dataframe_filtered.sort_values('distance')
sort_by_distance

name           categories  \
2                               Subway       Sandwich Place   
4   Akashiro Japanese Restaurant & Bar     Sushi Restaurant   
1        Terrace Restaurant at Elmwood  American Restaurant   
8                                 JOEY  American Restaurant   
3               The Senator Restaurant                Diner   
7                     Adega Restaurant           Restaurant   
9      Kyoto House Japanese Restaurant     Sushi Restaurant   
5  Richtree Natural Market Restaurants           Restaurant   
0      Hemispheres Restaurant & Bistro  American Restaurant   
6        Hong Shing Chinese Restaurant   Chinese Restaurant   

               address        lat        lng  \
2      10 Dundas St. E  43.656260 -79.380920   
4        220 Yonge St.  43.655965 -79.380541   
1                  NaN  43.656101 -79.381480   
8     1 Dundas St West  43.656094 -79.381878   
3  249 Victoria Street  43.655641 -79.379199   
7            33 Elm St  43.657519 -79.383462   
9  143 Dundas St. West  43.655381 -79.385270   
5        14 Queen St W  43.652614 -79.380231   
0  110 Chestnut Street  43.654884 -79.385931   
6      195 Dundas St W  43.654925 -79.387089   

                                      labeledLatLngs  distance postalCode  cc  \
2  [{'label': 'display', 'lat': 43.65625998968890...         6    M5B 2G9  CA   
4  [{'label': 'display', 'lat': 43.65596504293232...        49        NaN  CA   
1  [{'label': 'display', 'lat': 43.65610066909799...        51        NaN  CA   
8  [{'label': 'display', 'lat': 43.6560936540828,...        81    M5G 1Z3  CA   
3  [{'label': 'display', 'lat': 43.65564091455335...       157    M5B 1T8  CA   
7  [{'label': 'display', 'lat': 43.65751886446546...       244    M5G 1H1  CA   
9  [{'label': 'display', 'lat': 43.65538110598594...       365        NaN  CA   
5  [{'label': 'display', 'lat': 43.65261436174172...       416    M5H 3X4  CA   
0  [{'label': 'display', 'lat': 43.65488413420439...       434    M5G 1R3  CA   
6  [{'label': 'display', 'lat': 43.65492521335936...       520    M5G 1C7  CA   

      city state country                                   formattedAddress  \
2  Toronto    ON  Canada  [10 Dundas St. E (at Yonge St.), Toronto ON M5...   
4  Toronto    ON  Canada  [220 Yonge St. (in Toronto Eaton Centre Urban ...   
1      NaN   NaN  Canada                                           [Canada]   
8  Toronto    ON  Canada     [1 Dundas St West, Toronto ON M5G 1Z3, Canada]   
3  Toronto    ON  Canada  [249 Victoria Street (btwn Dundas St E and Shu...   
7  Toronto    ON  Canada  [33 Elm St (at Bay St), Toronto ON M5G 1H1, Ca...   
9  Toronto    ON  Canada          [143 Dundas St. West, Toronto ON, Canada]   
5  Toronto    ON  Canada        [14 Queen St W, Toronto ON M5H 3X4, Canada]   
0  Toronto    ON  Canada  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
6  Toronto    ON  Canada  [195 Dundas St W (at University Ave), Toronto ...   

                            crossStreet                        id  
2                          at Yonge St.  4c6abb8c2c29d13ad2720e41  
4  in Toronto Eaton Centre Urban Eatery  4fbbb3ace4b01e39077b4d55  
1                                   NaN  4edbccaaf9abeea3db5268df  
8                                   NaN  4df909dfe4cd2129701c0690  
3        btwn Dundas St E and Shuter St  4ad7929cf964a520500c21e3  
7                             at Bay St  4ad4c060f964a52070f720e3  
9                                   NaN  4ae4b055f964a520229d21e3  
5                                   NaN  4b295e10f964a520ba9d24e3  
0                                   NaN  4ad4c05ff964a52048f720e3  
6                     at University Ave  4b2027b5f964a520f82d24e3

In [12]:
# Hide the index
sort_by_distance.style.hide_index()

#### Explore Restaurants using their ID

In [13]:
# Making calls to Foursquare API to get 'given values'
# The Senator Restaurant would be our pilot call
# We then get the Rating, Price, Tier and Currency for others

In [14]:
venue_id = '4ad7929cf964a520500c21e3' # ID of The Senator Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4ad7929cf964a520500c21e3?client_id=2IFJ4OSQ4YRJ4ZF2BZZQXUNDJSNK4F4JHIFZ2RUVKVZ2FPQH&client_secret=FFJNCIUYMDQFICJLZMSZATWNHBBZCTCCZLRLUU1Z3JYSXNIS&v=20180604'

In [15]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

# To simplify, output was cleared

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'price', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'venuePage', 'reasons', 'description', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'popular', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4ad7929cf964a520500c21e3',
 'name': 'The Senator Restaurant',
 'contact': {'phone': '4163647517',
  'formattedPhone': '(416) 364-7517',
  'twitter': 'thesenatorto',
  'facebook': '566290650069193',
  'facebookUsername': 'thesenatorto',
  'facebookName': 'The Senator'},
 'location': {'address': '249 Victoria Street',
  'crossStreet': 'btwn Dundas St E and Shuter St',
  'lat': 43.65564091455335,
  'lng': -79.37919882575557,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.65564091455335,
    'lng': -79.37919882575557}],
  'postalCode': 'M5B 1T8',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['249 Victoria Street (btwn Dundas St E and Shuter St)',
   'Toronto ON M5B 1T8',
   'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/the-senator-restaurant/4ad7929cf964a520500c21e3',
 'categories': [{'id': '4bf58dd8d48988d147941735',
   'name': 'Diner',
   'pluralName': 'Diners',
   'shortName': 'Diner',
   'icon': {'prefix': 'h

In [16]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.9


In [17]:
try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')    

{'tier': 2, 'message': 'Moderate', 'currency': '$'}


In [18]:
# Now we leverage the Foursquare location data to the fullest
# let's get the Rating, Price, Tier and Currency for other restaurants


venue_id = '4c6abb8c2c29d13ad2720e41' # Subway
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    
    
venue_id = '4fbbb3ace4b01e39077b4d55' # Akashiro Japanese Restaurant & Bar
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.') 
    
        
venue_id = '4edbccaaf9abeea3db5268df' # Terrace Restaurant at Elmwood
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
venue_id = '539c6f13498e06f4cc765165' # The Elm Tree Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    
    
venue_id = '5a8b24b2018cbb6c4ca70dae' # Hendricks Restaurant & Bar
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    

venue_id = '4ae4b055f964a520229d21e3' # Kyoto House Japanese Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')
    

venue_id = '52a7ae41498eed3af4d0a3fa' # Yueh Tung Chinese Restaurant
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')

venue_id = '4b295e10f964a520ba9d24e3' # Richtree Natural Market Restaurants
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')


venue_id = '4ad4c05ff964a52048f720e' # Hemispheres Restaurant & Bistro
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)

result = requests.get(url).json()
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

try:
    print(result['response']['venue']['price'])
except:
    print('No Details.')


5.7
{'tier': 1, 'message': 'Cheap', 'currency': '$'}
5.4
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
This venue has not been rated yet.
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
8.3
No Details.
6.1
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
5.9
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
7.8
{'tier': 1, 'message': 'Cheap', 'currency': '$'}
6.2
{'tier': 2, 'message': 'Moderate', 'currency': '$'}
This venue has not been rated yet.
No Details.


##### Dataset Results above were extracted to csv format

In [19]:
# Let's get the csv file to create new dataframe
df_dtres = pd.read_csv('DT_Res_final.csv')
df_dtres

Restaurant Rating      Price
0                               Subway     5.7     Cheap
1   Akashiro Japanese Restaurant & Bar     5.4  Moderate
2        Terrace Restaurant at Elmwood      NR  Moderate
3               The Senator Restaurant     7.9  Moderate
4              The Elm Tree Restaurant     8.3        ND
5           Hendricks Restaurant & Bar     6.1  Moderate
6      Kyoto House Japanese Restaurant     5.9  Moderate
7         Yueh Tung Chinese Restaurant     7.8     Cheap
8  Richtree Natural Market Restaurants     6.2  Moderate
9      Hemispheres Restaurant & Bistro      NR        ND

##### New Dataframe: Rating and Price

In [20]:
# Look for word 'Cheap'
df_word=df_dtres[df_dtres['Price'].str.contains('Cheap')].reset_index(drop=True)

# Hide the index
df_word.style.hide_index()

### Table 1

In [21]:
# Look for word 'Moderate'
df_word=df_dtres[df_dtres['Price'].str.contains('Moderate')].reset_index(drop=True)

# Hide the index
df_word.style.hide_index()

### Table 2

<br>
<br>

## "Restaurant Guide App" (Two-Factor Approach):
  **Virtually**, Table 1 and 2 shows enough data without overwhelming our targeted audience. Cuisine Connoisseurs, Food Enthusiast and Bloggers, who constantly use restaurant guide app for quick venue information will appreciate this minimalistic comparison. Stakeholders has to consider the two-factor approach for improvement of consumer experience. Developers then can collaborate and utilize the datasets to come up with ways to show the Tables as part of the app. Sometimes, less can influence significant decisions but not to sacrifice functionality. Here, the Rating and Price is sufficient to recommend a venue. Our tables shows us, " Yueh Tung Chinese Restaurant and The Senator Restaurant" as our top picks.

<br>
<br>
<br>

#### Final Deliverables 

As per course instructions, The Final Deliverables will be shown in another Notebook for the **Second Week / Course Week 5**. The components supporting our "two-factor approach" will be further explain in the Methodology, Analysis, Results and Discussion, Conclusion sections. Included, will be the full report and presentation, and the final edited Notebook links to Github.